tests/devices/test_null_qubit.py

```diff
@@ -0,0 +1,1951 @@
+        greater than the target wire."""
+        state_out = method(self.state, axes=[2, 1, 0])
+        assert state_out == None
+
+    @pytest.mark.parametrize("method", three_qubit_ops)
+    def test_apply_three_qubit_op_controls_split(self, method, inverse):
+        """Test if the application of three qubit operations is correct when one control wire is smaller
+        and one control wire is greater than the target wire."""
+        state_out = method(self.state, axes=[3, 1, 2])
+        assert state_out == None
+
+    single_qubit_ops_param = [
+        (dev._apply_phase, [1.0]),
+    ]
+
+    @pytest.mark.parametrize("method,par", single_qubit_ops_param)
+    def test_apply_single_qubit_op_(self, method, par, inverse):
+        """Test if the application of single qubit operations (with parameter) is correct."""
+        state_out = method(self.state, axes=[1], parameters=par, inverse=inverse)
+        assert state_out == None
+
+
+class TestStateInitialization:
+    """Unit tests for state initialization methods"""
+
+    def test_state_vector_full_system(self, mocker):
+        """Test applying a state vector to the full system"""
+        state_wires = qml.wires.Wires(["a", "b", "c"])
+        dev = NullQubit(wires=state_wires)
+        state = np.array(
+            [
+                1 / math.sqrt(204),
+                2 / math.sqrt(204),
+                3 / math.sqrt(204),
+                4 / math.sqrt(204),
+                5 / math.sqrt(204),
+                6 / math.sqrt(204),
+                7 / math.sqrt(204),
+                8 / math.sqrt(204),
+            ]
+        )
+
+        spy = mocker.spy(dev, "_scatter")
+        dev._apply_state_vector(state=state, device_wires=state_wires)
+
+        assert dev._state == None
+        spy.assert_not_called()
+
+    def test_basis_state_full_system(self, mocker):
+        """Test applying a state vector to the full system"""
+        state_wires = qml.wires.Wires(["a", "b", "c"])
+        dev = NullQubit(wires=state_wires)
+        state = np.array(
+            [
+                1 / math.sqrt(204),
+                2 / math.sqrt(204),
+                3 / math.sqrt(204),
+                4 / math.sqrt(204),
+                5 / math.sqrt(204),
+                6 / math.sqrt(204),
+                7 / math.sqrt(204),
+                8 / math.sqrt(204),
+            ]
+        )
+
+        spy = mocker.spy(dev, "_scatter")
+        dev._apply_basis_state(state=state, wires=state_wires)
+
+        assert dev._state == None
+        spy.assert_not_called()
+
+
+class TestOpCallDirect:
+    """Tests the operation call statistics with direct calls to special methods."""
+
+    state = np.arange(2**4, dtype=np.complex128).reshape((2, 2, 2, 2))
+    dev = qml.device("null.qubit", wires=4)
+
+    single_qubit_ops = [
+        (dev._apply_x, {"PauliX": 1}),
+        (dev._apply_y, {"PauliX": 1, "PauliY": 1}),
```

Comment:
It feels a bit odd that the operation call dictionary keeps growing across the test cases as it makes the cases be a bit less readable and mutates the state of a single device stored as a class attribute. Do we surely want to pursue this approach?

Reply:
I simplified this part. Now the dictionary does not persist, I'm setting the device as a fixture. 
Thank you for your suggestion.